### PR TITLE
Fixed Split Strings in Brave Wallet

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Fixed strings through out the Brave Wallet that were not appropriately set up to be translated correctly.

Resolves <https://github.com/brave/brave-browser/issues/19217>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by bbondy** [comment]
Needs some lint fixes, `npm run format` should fix, then you can verify with `npm run lint`

**Comment by Douglashdaniel** [comment]
> Needs some lint fixes, `npm run format` should fix, then you can verify with `npm run lint`

Fixed lint errors! 👍

**Review by petemill — APPROVED** [review]
Good improvement. I left some comments about string descriptions and examples, but I'll leave it up to you as they might be ok either way. Sometimes translators can misunderstand and the translations end up incorrect so it's a bit of an art form.

FYI for the future if you get more complicated strings with multiple replacements I've seen some helpers floating around (perhaps not yet in a common location and are project specific, like Rewards) that can do more elegant calls such as `getString('stringKey', { $1: 'myString', $2: <MyReactComponent /> })`

File: components/brave_wallet_ui/components/desktop/wallet-onboarding/welcome/index.tsx [hunk 1]
```diff
@@ -37,6 +37,18 @@ function OnboardingWelcome (props: Props) {
...
+    const text = getLocale('braveWalletCryptoWalletsDescriptionTwo')
+    const { beforeTag, duringTag, afterTag } = splitStringForTag(text)
+    return (
+      <CryptoWalletsAlertDescription>
+        {beforeTag}
+        <SettingsButton onClick={onClickSettings}>{duringTag}</SettingsButton>
+        {afterTag}
+      </CryptoWalletsAlertDescription>
+    )
+  }, [])
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/wallet-onboarding/welcome/index.tsx]: Good call! This has been updated!

File: components/resources/wallet_strings.grdp [hunk 2]
```diff
@@ -202,15 +202,15 @@
   <message name="IDS_BRAVE_WALLET_SWAP_LIMIT" desc="Swap tab limit button">Limit</message>
   <message name="IDS_BRAVE_WALLET_SWAP_PRICE_IN" desc="Swap tab price label">Price in</message>
   <message name="IDS_BRAVE_WALLET_BUY_TITLE" desc="Buy tab faucet button">Test faucet</message>
-  <message name="IDS_BRAVE_WALLET_BUY_DESCRIPTION" desc="Buy tab faucet description">Get Ether from a faucet for</message>
+  <message name="IDS_BRAVE_WALLET_BUY_DESCRIPTION" desc="Buy tab faucet description">Get Ether from a faucet for <ph name="SELECTED_NETWORK">$1</ph></message>
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/resources/wallet_strings.grdp]: Thank you for the recommendation! 
I've added examples to all the strings that need it and a clearer description.

File: components/resources/wallet_strings.grdp [hunk 3]
```diff
@@ -244,38 +244,35 @@
...
+  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_CONNECTED" desc="Connect Hardware panel connected status"><ph name="DEVICE_NAME">$1</ph> connected</message>
+  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_DISCONNECTED" desc="Connect Hardware panel disconnected status"><ph name="DEVICE_NAME">$1</ph> disconnected</message>
   <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_INSTRUCTIONS" desc="Connect Hardware panel instructions button">Instructions</message>
-  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_CONNECT" desc="Connect Hardware panel title">Connect your</message>
-  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_CONFIRMATION" desc="Connect Hardware panel confirmation description">Hardware wallet requires transaction confirmation.</message>
+  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_CONNECT" desc="Connect Hardware panel title">Connect your <ph name="DEVICE_NAME">$1</ph></message>
+  <message name="IDS_BRAVE_WALLET_CONNECT_HARDWARE_PANEL_CONFIRMATION" desc="Connect Hardware panel confirmation description">Hardware wallet requires transaction confirmation. <ph name="DEVICE_NAME">$1</ph></message>
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/resources/wallet_strings.grdp]: Examples added!

**Review by bbondy — APPROVED** [review]

**Pull request merged**